esm/src/api/controllers/demo.controller.js

```diff
@@ -0,0 +1,201 @@
+    console.log(
+      chalk.greenBright(
+        `\nGET request successful! \n\nRunning at http://localhost:3000/${routeName}/\n`
+      )
+    );
+  } catch (err) {
+    res.status(500).json({
+      error: `${err}`,
+    });
+    console.log(chalk.redBright(`\nError retriving ${item}s: ${err}\n`));
+  }
+};
+
+const createDemoItemController = async (req, res) => {
+  try {
+    let doc = await createDemoItemService(req.body);
+    res.status(201).json({
+      message: `${item} created successfully!`,
+      newItem: {
+        _id: doc._id,
+        name: doc.name,
+        age: doc.age,
+        request: {
+          type: "GET",
+          url: `http://localhost:3000/${routeName}/${doc._id}`,
+        },
+      },
+    });
+    console.log(
+      chalk.greenBright(
+        `\n${item} CREATED successfully! \n\nCreated ${item} url: http://localhost:3000/${routeName}/${doc._id}\n`
+      )
+    );
+  } catch (err) {
+    res.status(500).json({
+      error: `${err}`,
+    });
+    console.log(chalk.redBright(`\nError saving ${item}: ${err}\n`));
+  }
+};
+
+/*=============================
+  operations for /demo/demoId
+==============================*/
+
+const getOneDemoItemController = async (req, res, next) => {
+  try {
+    let doc = await getOneDemoItemService(req.params.demoId);
+    if (doc) {
+      res.status(200).json({
+        _id: doc._id,
+        name: doc.name,
+        age: doc.age,
+        request: {
+          type: "GET",
+          description: `Url link to all ${item}s`,
+          url: `http://localhost:3000/${routeName}/`,
+        },
+      });
+      console.log(
+        chalk.greenBright(
+          `\nGET request successful! \n\n${item} url: http://localhost:3000/${routeName}/${doc._id}\n`
+        )
+      );
+    } else {
+      console.log(chalk.redBright("\nNo record found for provided ID\n"));
+      return res.status(404).json({
+        message: "No record found for provided ID",
+      });
+    }
+  } catch (err) {
+    res.status(500).json({
+      message: "Invalid ID",
+      error: `${err}`,
+    });
+    console.log(chalk.redBright(`\nError retriving ${item}: ${err}\n`));
+  }
+};
+
+const deleteDemoItemController = async (req, res, next) => {
+  try {
+    let doc = await deleteDemoItemService(req.params.demoId);
+    console.log(chalk.greenBright(`\n${item} DELETED successfully!\n`));
+    res.status(200).json({
+      message: `${item} deleted successfully!`,
+      request: {
+        type: "POST",
+        description: "Url link to make post request to",
+        url: `http://localhost:3000/${item}/`,
+        body: {
+          name: "String",
+          age: "Number",
+        },
+      },
+    });
+  } catch (err) {
+    res.status(500).json({
+      message: `Error deleting ${item}`,
+      error: `${err}`,
+    });
+    console.log(chalk.redBright(`\nError deleting ${item}: ${err}\n`));
+  }
+};
+
+const partialUpdateDemoItemController = async (req, res, next) => {
```

Comment:
Nice work @samuko-things 

"partialUpdate" is a bit general. It will be better to be more specific.

If you notice in the service we are interacting with the database using the `Demo.updateOne()` method.
So it will be better to replace `partialUpdateDemoItemController` with `updateOneDemoItemPropertyValueController` in the function name(s).

(also remember to do this for services, routes and any other place you used "partialUpdate")

It will help the dev using it to quickly know what the function does just by looking at the function name without thinking too much.
These questions should give insight to why it's better to so this:
- What is being updated? => "propertyValue"
- How many of it are we updating? => "updateOne"

Youtube video reference (for patch request updating just one): https://youtu.be/WDrU305J1yw?t=1855